integrationTests/vm/arwen/arwenvm/execute_via_blockchainhook_test.go

```diff
@@ -0,0 +1,79 @@
+package arwenvm
+
+import (
+	"encoding/hex"
+	"math/big"
+	"testing"
+
+	"github.com/ElrondNetwork/elrond-go/integrationTests"
+	"github.com/ElrondNetwork/elrond-go/process/factory"
+	"github.com/ElrondNetwork/elrond-vm-common/txDataBuilder"
+	"github.com/ElrondNetwork/wasm-vm/mock/contracts"
+	"github.com/ElrondNetwork/wasm-vm/testcommon"
+	test "github.com/ElrondNetwork/wasm-vm/testcommon"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_ExecuteOnDestCtx_BlockchainHook(t *testing.T) {
+
+	net := integrationTests.NewTestNetworkSized(t, 1, 1, 1)
+	net.Start()
+	net.Step()
+
+	net.CreateWallets(2)
+	net.MintWalletsUint64(100000000000)
+	ownerOfParent := net.Wallets[0]
+
+	fakeVMType, _ := hex.DecodeString("abab")
+
+	node := net.NodesSharded[0][0]
+	parentAddress, _ := GetAddressForNewAccount(t, net, node)
+	childAddress, _ := GetAddressForNewAccountWithVM(t, net, node, fakeVMType)
+
+	testConfig := &testcommon.TestConfig{
+		ParentBalance: 20,
+		ChildBalance:  10,
+
+		GasProvided:        2_000_000,
+		GasProvidedToChild: 1_000_000,
+		GasUsedByParent:    400,
+	}
+
+	defaultVM, _ := node.VMContainer.Get(factory.ArwenVirtualMachine)
+	err := node.VMContainer.Add(fakeVMType, defaultVM)
```

Comment:
so we essentially add another VM which is actually the same pointer of the regular VM used in most tests from this package.
This is fine for these tests to check if the calls are being made. In a production setup, we will need to prevent this, explained below.